SimCore/src/SimCore/Generators/LHEPrimaryGenerator.cxx

```diff
@@ -1,53 +1,35 @@
-/**
- * @file LHEPrimaryGenerator.cxx
- * @brief Implementation file for LHEPrimaryGenerator
- * @author Jeremy McCormick, SLAC National Accelerator Laboratory
- * @author Tom Eichlersmith, University of Minnesota
- */
-
 #include "SimCore/Generators/LHEPrimaryGenerator.h"
 
-// Geant4
-#include "G4Event.hh"
-#include "G4IonTable.hh"
-#include "G4PhysicalConstants.hh"
-#include "G4RunManager.hh"
-#include "G4SystemOfUnits.hh"
-
-// LDMX
-#include "Framework/Configure/Parameters.h"
-#include "Framework/Exception/Exception.h"
-#include "SimCore/G4User/UserPrimaryParticleInformation.h"
-
 namespace simcore {
 namespace generators {
 
 LHEPrimaryGenerator::LHEPrimaryGenerator(
     const std::string& name, const framework::config::Parameters& parameters)
     : PrimaryGenerator(name, parameters) {
   file_path_ = parameters.getParameter<std::string>("filePath");
-  reader_ = new simcore::lhe::LHEReader(file_path_);
+  reader_ = std::make_unique<simcore::lhe::LHEReader>(file_path_);
```

Comment:
since the reader is created in the constructor and we don't need to re-create it some other time, I would just have the reader be an object instead of a pointer.

```
// header
// file_path_ needs to be declared before reader_ so that it is initialized first
std::string file_path_;
LHEReader reader_;

// impl
LHEPrimaryGenerator::LHEPrimaryGenerator(
    const std::string& name, const framework::config::Parameters& parameters)
    : PrimaryGenerator(name, parameters), file_path_{parameters.get("filePath")}, reader_{file_path_} {}
```